.github/workflows/deploy-to-remote-repository.yml

```
@@ -0,0 +1,105 @@
+name: Sync to remote repository
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        default: "ubuntu-latest"
+        required: false
+        type: string
+      remote_repo:
+        description: 'Remote repository URL'
+        required: true
+        type: string
+      remote_branch:
+        description: 'Remote branch to clone and push'
+        required: false
+        type: string
+      base_directory:
+        description: 'Base directory for rsync'
+        required: false
+        default: '.'
+        type: string
+      destination_directory:
+        description: 'Destination directory for rsync in remote repository'
+        required: false
+        default: '.'
+        type: string
+      exclude_list:
+        description: 'Comma-separated list of files and directories to exclude from sync'
+        required: false
+        default: '.git, .gitmodules'
+        type: string
+    secrets:
+      REMOTE_REPO_SSH_KEY:
+        description: 'SSH key for remote repository'
+        required: true
+
+jobs:
+  deploy:
+    runs-on: ${{ inputs.os }}
+    name: "Syncing ${{ inputs.base_directory }} to ${{ inputs.remote_repo }}@${REMOTE_BRANCH}: ${{ inputs.destination_directory }}"
+
+    env:
+      GIT_SSH_COMMAND: ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -i ~/.ssh/private_key
+      REMOTE_REPO_DIR: /tmp/remote_repo
+      REMOTE_BRANCH: ${{ inputs.remote_branch != '' && inputs.remote_branch || github.ref_name }}
+      SSH_KEY: ${{ secrets.REMOTE_REPO_SSH_KEY }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
```

Could each action have a line after it for better code reading?

```suggestion
      - name: Checkout code
        uses: actions/checkout@v3

```

---

yeah!